src/inmanta/env.py

```diff
@@ -1281,3 +1297,64 @@ class VenvActivationFailedError(Exception):
     def __init__(self, msg: str) -> None:
         super().__init__(msg)
         self.msg = msg
+
+
+@dataclass
+class VenvSnapshot:
```

Comment:
Nice! Well, once you accept the horrible things we do here, but that's an established fact.